### PR TITLE
Use bcp47 language code for AFF

### PR DESCRIPTION
**Pull request by wurzka**
In https://github.com/SAP/abap-file-formats/issues/582 it was decided that we use BCP47 language tags for ABAP file formats. With this pull request, it is enabled for interfaces in abapGit.

### CODE REVIEW

**Comment by albertmink** [comment]
This change affects both experimental features `AFF` and `TRANS` afaik. TRANS because the language is also represented in file name, see https://github.com/SAP/abap-file-formats/blob/main/docs/specification.md#language
Be aware that the PR  aims to align the experimental features `AFF` to current changes in https://github.com/SAP/abap-file-formats/issues/582 not so much the feature `TRANS`. Meaning the 'integration' test here
https://github.com/abapGit/abapGit/blob/3bfb2be6ab270103c816768565440f5c8aac9fc0/src/objects/zcl_abapgit_objects.clas.testclasses.abap#L231
will fail for any BCP47 != char[2]
So the is a potential race (incompatibility) of experimental feature flags, which might be worth to be addressed in a separate PR.

File: src/utils/zcl_abapgit_convert.clas.locals_imp.abap [hunk 1]
```diff
@@ -115,3 +115,678 @@ CLASS lcl_out IMPLEMENTATION.
...
+  METHOD fill_language_mappings.
+    DATA line TYPE ty_language_mapping.
+
+    line-bcp47_code = 'af'.
+    line-sap1_code = 'a'.
+    APPEND line TO language_mappings.
+    CLEAR line.
+
+    line-bcp47_code = 'sq'.
+    line-sap1_code = '뽑'.
```
Comment (wurzka) [inline src/utils/zcl_abapgit_convert.clas.locals_imp.abap:196]: @larshp Do you have an idea here? We need these characters as they are used as SAP1 Language code. However, abaplint complains that they are non 7 bit ascii characters. Can we skip this test here somehow?

Comment (pokrakam) [reply src/utils/zcl_abapgit_convert.clas.locals_imp.abap:196]: @wurzka it's possible, but would be better to represent it in UTF. 
`line-sap1_code = cl_abap_conv_in_ce=>uccp( 'BF51' ).` 
(found by pasting your character into https://www.coderstool.com/unicode-text-converter)

Comment (wurzka) [reply src/utils/zcl_abapgit_convert.clas.locals_imp.abap:196]: Great. Thanks @pokrakam. Will be fixed with the next commit

File: src/utils/zcl_abapgit_convert.clas.abap [hunk 2]
```diff
@@ -183,6 +199,92 @@ CLASS zcl_abapgit_convert IMPLEMENTATION.
...
+        TRY.
+            re_lang_sap1 = lcl_bcp47_language_table=>bcp47_to_sap1( im_lang_bcp47 ).
+          CATCH zcx_abapgit_exception.
+
+            CREATE OBJECT lv_regex EXPORTING pattern = `[A-Z0-9]{2}`.
+            lv_abap_matcher = lv_regex->create_matcher( text = im_lang_bcp47 ).
+
+            IF abap_true = lv_abap_matcher->match( ).
+              "Fallback try to convert from SAP language
+              lv_sap1_converter_class = 'CL_I18N_LANGUAGES'.
```
[the remark this answers is not in the document]
Comment (wurzka) [reply src/utils/zcl_abapgit_convert.clas.abap]: Thanks @mbtools. Fixed it.

**Comment by schneidermic0** [comment]
@mbtools Check "abaplint / cross check" reports an issue in one of your repos (Marc-Bernard-Tools/abaplint-Ext-for-abapGit; V2). Is this an issue?

**Comment by mbtools** [comment]
The v2 branch in my repo is wip. You can ignore it.

**Pull request merged**